python/cudf/cudf/core/index.py

```diff
@@ -2744,6 +2748,214 @@ def categories(self):
         return self._values.cat().categories
 
 
+def interval_range(
+    start: int = None,
+    end: int = None,
```

Comment:
I think these need to handle datetime-likes as well, where it's okay if they don't in this PR, but we should probably raise a `NotImplementedError` if we don't implement them. That being said moving to `column.arange` should likely help to handle datetime-likes as well.

Reply:
Sounds good, I've switched to `column.arange` and running some tests with datetime-likes and strings. If I'm running into too many failures I'll raise as suggested :D

Reply:
`column.arange` doesn't seem to handle datetime-likes either. I'm raising a `NotImplementedError` for the moment, and will work on handling them in the next PR!